cli/command/image/build.go

```diff
@@ -247,6 +253,66 @@ func runBuild(dockerCli *command.DockerCli, options buildOptions) error {
 		buildCtx = replaceDockerfileTarWrapper(ctx, buildCtx, relDockerfile, translator, &resolvedTags)
 	}
 
+	// write ctx tar to disk, to make it readable for Image Reference extraction
+	originCtx := buildCtx
+	defer originCtx.Close()
+	readbackTar, err := ioutil.TempFile("", "docker-ctx")
```

Comment:
The problem is that the wrapper's transformation is done while uploading the request body. I need to extract the referenced registries before sending the request to include the requested auth configs in the sent request. So I need the context to be seekable (to read it twice).